### PR TITLE
add privileged and parameters definition to AppDefinition json schema

### PR DESCRIPTION
**Pull request by tamarrow**
DCOS-1345

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
Related: #1185

**Comment by ConnorDoyle** [comment]
Although not necessarily required to accept this change, how do we validate conformance between the JSON schema and the actual JSON formatters used to render response bodies?

There's a Java project to do this called [json-schema-validator](https://github.com/fge/json-schema-validator}.  The README states it's looking for a new maintainer, but maybe that's not a huge problem if it's only used in the tests.

It seems like it wouldn't be too much work to render the JSON for a number of AppDefinition instances and ensure they validate against the in-tree schema.  This could increase confidence / awareness when the API changes, both intentionally and otherwise.

**Comment by tamarrow** [comment]
In the future the best way is to go directly from the source to generate the schema. We could even have the api call generate the schema so we don't save state that could get outdated. And yes, change the tests to use this schema to validate. 

Where is the AppDefinition defined in Marathon? There seems to be a few sources of truth that are used in the codebase.

**Comment by ConnorDoyle** [comment]
The JSON rendering is currently defined in `mesosphere.marathon.api.v2.json.Formats.scala`.

**Comment by tamarrow** [comment]
@ConnorDoyle I added the json-schema-validator project and used it to verify the json schema with sample apps in the AppDefinitionValidator tests. Using it I found some fields we missed!

File: project/build.scala [hunk 1]
```diff
@@ -224,6 +226,7 @@ object Dependency {
   val hadoopCommon = "org.apache.hadoop" % "hadoop-common" % V.Hadoop excludeAll(excludeMortbayJetty, excludeJavaxServlet)
   val beanUtils = "commons-beanutils" % "commons-beanutils" % "1.9.2"
   val scallop = "org.rogach" %% "scallop" % V.Scallop
+  val jsonSchemaValidator = "com.github.fge" % "json-schema-validator" % V.JsonSchemaValidator
```
Comment (ConnorDoyle) [inline project/build.scala]: Verified Apache 2.0 compliant: https://github.com/fge/json-schema-validator/blob/master/LICENSE#L13

**Comment by ConnorDoyle** [comment]
Awesome, thanks @tamarrow!
If it's not too much trouble could we also validate the JSON in https://github.com/mesosphere/marathon/blob/master/src/test/scala/mesosphere/marathon/state/AppDefinitionTest.scala?

**Comment by ConnorDoyle** [comment]
I already feel a lot more confident in the schema correctness though :sunglasses:

**Comment by kolloch** [comment]
Looks good to me. Thank you.

**Comment by ConnorDoyle** [comment]
@kolloch we were waiting to merge this until additional tests were added.  Please read the outstanding comments before merging.

**Comment by ConnorDoyle** [comment]
The follow-up PR is here: #1547

**Comment by kolloch** [comment]
Hi @ConnorDoyle ,

I actually wanted this as a basis for my schema extensions which is why I merged it. I think it was already an improvement over the prior state.

Anyways, sorry if I caused you any trouble and thanks for the work.

**Pull request merged**